client/src/route/auth/PrivateRoute.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { ReactNode } from 'react';
+import { Navigate } from 'react-router-dom';
+import { useAuth } from 'react-oidc-context';
+import WaitNavigateAndReload from './WaitAndNavigate';
+
+interface PrivateRouteProps {
+  children: ReactNode;
+}
+
+const PrivateRoute: React.FC<PrivateRouteProps> = ({ children }) => {
+  const auth = useAuth();
+  const [isInitialFetchDone, setIsInitialFetchDone] = React.useState(false);
+  let returnJSX;
+
+  React.useEffect(() => {
+    if (auth.isAuthenticated && !isInitialFetchDone) {
```

Comment:
This code is quite similar to setandgetUsername in `util/auth/Authentication.ts`. Can it be deduplicated?

Reply:
We are just using this part for storing of username, we have now moved it to getAndSetUsername in util/auth/Authentication.ts.